railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -120,24 +120,25 @@ def config
       empty_directory "config"
 
       inside "config" do
-        template "routes.rb"
+        template "routes.rb" unless options[:updating]
         template "application.rb"
-        template "environment.rb"
-        template "cable.yml" unless options[:skip_action_cable]
-        template "puma.rb"   unless options[:skip_puma]
-        template "spring.rb" if spring_install?
-        template "storage.yml" unless skip_active_storage?
+        template "environment.rb" unless options[:updating]
```

Comment:
I want to keep this file in the upgrading task. It is not often that we change then, but when we do it will certainly break the application if you don't upgrade.